powershell_completions.go

```diff
@@ -11,90 +8,253 @@ import (
 
-func escapeStringForPowerShell(s string) string {
-	return strings.Replace(s, "'", "''", -1)
+        ## get the paths from Get-ChildItem
+        __%[1]s_debug "ShellCompDirectiveFilterFileExt"
+
+        $Values | ForEach-Object {
+            __%[1]s_debug "Filter: $($_.Name)"
+            Get-ChildItem -Recurse -Name -File -Filter "*.$($_.Name)"
+        }
+        return
+    }
+
+    $Values = $Values | Where-Object {
+        # filter the result
+        $_.Name -like "$WordToComplete*"
+    } 
+
+    #Get the current mode
+    $Mode = (Get-PSReadLineKeyHandler | Where-Object {$_.Key -eq "Tab" }).Function
+    __%[1]s_debug "Mode: $Mode"
+
+
+
+    $Values | ForEach-Object {
+
+        # store temporay because switch will overwrite $_
+        $comp = $_
+
+
+        # Powershell supports three different completion modes
+        # - TabCompleteNext (default windows style - on each key press the next option is displayed)
+        # - Complete (works like bash)
+        # - MenuComplete (works like zsh)
+        # You set the mode with Set-PSReadLineKeyHandler -Key Tab -Function <mode>
+
+        # CompletionResult Arguments:
+        # 1) CompletionText text to be used as the auto completion result
+        # 2) ListItemText   text to be displayed in the suggestion list
+        # 3) ResultType     type of completion result
+        # 4) ToolTip        text for the tooltip with details about the object
+
+        switch ($Mode) {
+
+            # bash like
+            "Complete" {
+
+                if ($Values.Length -eq 1) {
+                    __%[1]s_debug "Only one completion left"
+
+                    # insert space after value
+                    [System.Management.Automation.CompletionResult]::new("$($comp.Name)$Space", "$($comp.Name)", 'ParameterValue', "$($comp.Description)")
+
+                } else {
+                    ##TODO Trim Description if to long
+
+                    # Add the proper number of spaces to align the descriptions
+                    while($comp.Name.Length -lt $Longest) {
+                        $comp.Name = $comp.Name + " "
+                    }
+
+                    [System.Management.Automation.CompletionResult]::new("$($comp.Name)  ($($comp.Description))", "$($comp.Name)  ($($comp.Description))", 'ParameterValue', "$($comp.Description)")
+                }
+             }
+
+            # zsh like
+            "MenuComplete" {
+                # insert space after value
+                # MenuComplete will automatically show the ToolTip of 
+                # the highlighted value at the bottom of the suggestions.
+                [System.Management.Automation.CompletionResult]::new("$($comp.Name)$Space", "$($comp.Name)", 'ParameterValue', "$($comp.Description)")
+            }
+
+            # TabCompleteNext and in case we get something unknown
+            Default {
+                # Like MenuComplete but we dont want to add a space here because 
```

Comment:
`don't` instead of `dont`

(it's my VSCode that's reporting me these typos)

Reply:
I also use VSCode, is that an extension? It seams like I need that one too.

Reply:
I use the extension "Code Spell Checker"